networkx/readwrite/nx_ete.py

```diff
@@ -0,0 +1,221 @@
+"""
+****
+ETE
+****
+Read and write NetworkX graphs in the New Hampshire Newick format.
+
+"ETE (Environment for Tree Exploration) is a Python programming toolkit
+that assists in the automated manipulation,
+analysis and visualization of phylogenetic trees.
+
+It uses the New Hampshire Newick format as one of the most widely used
+standard representation of trees in bioinformatics."
+
+See http://etetoolkit.org/ for documentation.
+
+Format
+------
+http://etetoolkit.org/docs/latest/tutorial/tutorial_trees.html#reading-and-writing-newick-trees
+
+"""
+
+__all__ = ["to_ete", "from_ete", "write_ete", "read_ete"]
+
+import networkx as nx
+from networkx.utils import open_file, deque
+
+
+def to_ete(G):
+    """Returns a ETE Tree from NetworkX arborescence [3]_ G.
+
+    ETE (Environment for Tree Exploration) is a Python programming toolkit
+    that assists in the automated manipulation,
+    analysis and visualization of phylogenetic trees [2]_.
+
+    It uses the New Hampshire Newick format as one of the most widely used
+    standard representation of trees in bioinformatics [1]_.
+
+
+    Parameters
+    ----------
+    G : NetworkX arborescence
+
+    Returns
+    -------
+    T : ETE Tree
+
+    Examples
+    --------
+    >>> G = nx.path_graph(42, create_using=nx.DiGraph)
+    >>> T = nx.to_ete(G)
+
+    References
+    ----------
+    .. [1] http://etetoolkit.org/docs/latest/tutorial/tutorial_trees.html#reading-and-writing-newick-trees
+    .. [2] http://etetoolkit.org/
+    .. [3] https://networkx.org/documentation/stable//reference/algorithms/tree.html#tree
+    """
+    if not nx.is_arborescence(G):
+        raise ValueError(f"Input graph {G} must be an arborescence.")
```

Comment:
```suggestion
        raise ValueError(f"Input graph G must be an arborescence.")
```
I would suggest dropping the formatted output, only because it will make the error message more verbose, but not necessarily any more clear. The current `__str__` for the graph class prints something like `Graph {name} with {num_nodes} nodes and {num_edges} edges`.